Jenkinsfile

```diff
@@ -0,0 +1,88 @@
+
+@Library('apm@current') _
+
+pipeline {
+  agent { label 'linux && immutable' }
+  environment {
+    REPO = 'golang-crossbuild'
+    BASE_DIR = "src/github.com/elastic/${env.REPO}"
+    NOTIFY_TO = credentials('notify-to')
+    PIPELINE_LOG_LEVEL = 'INFO'
+    HOME = "${env.WORKSPACE}"
+    DOCKER_REGISTRY_SECRET = 'secret/apm-team/ci/docker-registry/prod'
+    REGISTRY = 'docker.elastic.co'
+    STAGING_IMAGE = "${env.REGISTRY}/observability-ci"
+  }
+  options {
+    timeout(time: 2, unit: 'HOURS')
+    buildDiscarder(logRotator(numToKeepStr: '20', artifactNumToKeepStr: '20', daysToKeepStr: '30'))
+    timestamps()
+    ansiColor('xterm')
+    disableResume()
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+    rateLimitBuilds(throttle: [count: 60, durationName: 'hour', userBoost: true])
+    quietPeriod(10)
+  }
+  triggers {
+    issueCommentTrigger('(?i).*jenkins\\W+run\\W+(?:the\\W+)?tests(?:\\W+please)?.*')
+  }
+  stages {
+    stage('Checkout') {
+      steps {
+        deleteDir()
+        gitCheckout(basedir: BASE_DIR)
+        stash allowEmpty: true, name: 'source', useDefaultExcludes: false
```

Comment:
Nit: should `allowEmpty` really be set to `true`? Seems like subsequent stages are sure to fail if this stash is empty.

Reply:
```suggestion
        stash name: 'source', useDefaultExcludes: false
```